File: docs-examples/example-java/src/test/java/io/micronaut/configuration/rabbitmq/docs/consumer/acknowledge/bool/BooleanAckSpec.java
package io.micronaut.configuration.rabbitmq.docs.consumer.acknowledge.bool;

import io.micronaut.configuration.rabbitmq.AbstractRabbitMQTest;
import io.micronaut.context.ApplicationContext;
import org.junit.jupiter.api.Test;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

public class BooleanAckSpec extends AbstractRabbitMQTest {

    @Test
    void testAckingWithBoolean() {
        ApplicationContext applicationContext = startContext();

// tag::producer[]
ProductClient productClient = applicationContext.getBean(ProductClient.class);
productClient.send("message body".getBytes());
// end::producer[]

        ProductListener productListener = applicationContext.getBean(ProductListener.class);

        try {
            await().atMost(5, SECONDS).until(() ->
                    productListener.messageCount.get() == 2
            );
        } finally {
            applicationContext.close();
        }
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/rabbitmq/docs/consumer/acknowledge/bool/BooleanAckSpec.kt
package io.micronaut.configuration.rabbitmq.docs.consumer.acknowledge.bool;

import io.kotlintest.eventually
import io.kotlintest.seconds
import io.kotlintest.shouldBe
import io.micronaut.configuration.rabbitmq.AbstractRabbitMQTest
import org.opentest4j.AssertionFailedError

class BooleanAckSpec : AbstractRabbitMQTest({

    val specName = javaClass.simpleName

    given("An acknowledgement argument") {
        val ctx = startContext(specName)

        `when`("The messages are published") {
            val productListener = ctx.getBean(ProductListener::class.java)

            // tag::producer[]
            val productClient = ctx.getBean(ProductClient::class.java)
            productClient.send("body".toByteArray())
            // end::producer[]

            then("The messages are received") {
                eventually(10.seconds, AssertionFailedError::class.java) {
                    productListener.messageCount.get() shouldBe 2
                }
            }
        }

        ctx.stop()
    }
})